pkg/artifactory/resource_artifactory_virtual_debian_repository.go

```diff
@@ -29,12 +29,8 @@ func resourceArtifactoryDebianVirtualRepository() *schema.Resource {
 			MinItems: 0,
 			Computed: true,
 			Elem: &schema.Schema{
-				Type: schema.TypeString,
-				ValidateFunc: validation.StringInSlice([]string{
-					"bz2",
-					"lzma",
-					"xz",
-				}, false),
+				Type:             schema.TypeString,
+				ValidateDiagFunc: validateTypeSetStringList([]string{"bz2", "lzma", "xz"}),
```

Comment:
I don't believe you need this validator at all. I just sampled this with some other code and it compiles fine

```go
		"external_dependencies_patterns": {
			Type:     schema.TypeList,
			Optional: true,
			ForceNew: true,
			Elem: &schema.Schema{
				Type: schema.TypeString,
				ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice([]string{"slice"},false)),
			},
			RequiredWith: []string{"external_dependencies_enabled"},
			Description: "An allow list of Ant-style path patterns that determine which remote VCS roots Artifactory will " +
				"follow to download remote modules from, when presented with 'go-import' meta tags in the remote repository response. " +
				"By default, this is set to '**', which means that remote modules may be downloaded from any external VCS source.",
		},
```